### ISSUE TITLE
MeasureManager, ParameterManager and other intermediaries

### ISSUE THREAD

**Issue by matthieu-vergne** [jMetal/jMetal#74]
I just though about a thing last night. If I am correct, this would lead to further simplification by removing some interfaces and methods, but it is also a design choice, so better to get your opinion too.

Bascially, there is a recurrent pattern in the interfaces I have provided:
- something can be `Measurable` by providing a `MeasureManager` which will provide the `Measure` instances
- something can be `Parameterable` by providing a `ParameterManager` which will provide the `Parameter` instances

But more I think about it, and more I see these intermediaries (`XxxManager`) as useless. Not necessarily the implementations, which can be used to simplify the internal management of the measures/parameters/..., but introducing them at the conceptual level seems to me irrelevant.

For instance, if I take the measures, I would replace:
```
public interface Measurable {
  public MeasureManager getMeasureManager();
}
public interface MeasureManager {
  public Collection<Object> getMeasureKeys();
  public <T> PullMeasure<T> getPullMeasure(Object key);
  public <T> PushMeasure<T> getPushMeasure(Object key);
}
```
by:
```
public interface Measurable {
  public Collection<Object> getMeasureKeys();
  public <T> PullMeasure<T> getPullMeasure(Object key);
  public <T> PushMeasure<T> getPushMeasure(Object key);
}
```
thus removing the intermediary manager. Why? Because one can still use such a manager even if we don't ask for it, as well as not using one although we ask for it.

More in details, the usual case is to implement an algorithm, then inside the algorithm implements the set of measures and the measure manager to store them. But nothing forbids you to make the algorithm implements `MeasureManager` by itself, and for `getMeasureManager()` simply return `this`. At the opposite, assuming we remove this intermediary and directly require the algorithm to impleent the `getXxxMeasure(key)` methods, it can internally use a manager (which would also implements `Measurable`) to store the measures and simply call `manager.getXxxMeasure(key)` instead of managing it by itself.

In other words, while it makes sense to have implementations to simplify the work, introducing it in the root interfaces have no concrete effect. Based on that, the Occam's razor says that we should remove it, because:
- it reduces the number of interfaces (remove the intermediary),
- it reduces the verbosity of the code (replace `algorithm.getMeasureManager.getXxxMeasure(key)` by `algorithm.getXxxMeasure(key)`),
- yet it does not remove any feature.

Thus, such an intermediary becomes a facility (that we could put in an `util` package) and not a fundamental concept.
